### PR TITLE
Pack optimizations

### PR DESCRIPTION
**Pull request by tzulberti-jampp**

### CODE REVIEW

File: sharedbuffers/mapped_struct.py [hunk 1]
```diff
@@ -332,12 +361,18 @@ class WRAPPED:
     pass
 
 
+WRAP_MASK = cython.declare(object, 1L << 64)
```
Comment (tzulberti-jampp) [inline sharedbuffers/mapped_struct.py:364]: Are this constants defined here for any particular reason?

File: sharedbuffers/mapped_struct.py [hunk 2]
```diff
@@ -332,12 +361,18 @@ class WRAPPED:
...
+FLAG_MASK = cython.declare(object, 0xFL << 64)
+
+
 @cython.ccall
 def wrapped_id(obj):
     xid = shared_id(obj)
     if isinstance(xid, (int, long)):
         # We keep numeric ids numeric
-        xid |= 0xFL << 64
+        xid |= WRAP_MASK
```
Comment (tzulberti-jampp) [inline sharedbuffers/mapped_struct.py:375]: Shouldn't this be FLAG_MASK? or there was a bug about that?

File: sharedbuffers/mapped_struct.py [hunk 3]
```diff
@@ -441,72 +511,132 @@ def unpack_from(cls, buf, offs, idmap = None):
...
+                        buf[offs] = 'q'
+                    elif dtype == 'L':
+                        buf[offs] = 'Q'
+                    else:
+                        buf[offs] = dtype
+        else:
+            all_int = all_intlong = all_float = 1
+            minval = maxval = 0
+            for x in obj:
+                if all_int and type(x) is not int:
```
Comment (tzulberti-jampp) [inline sharedbuffers/mapped_struct.py:557]: Is this faster than checking `isinstance(x, int)`?

Comment (tzulberti-jampp) [inline sharedbuffers/mapped_struct.py:557]: Why don't just calculate the type once: `x_type = type(x)`?

File: sharedbuffers/mapped_struct.py (hunk 1 above)
Comment (klaussfreire) [reply sharedbuffers/mapped_struct.py:364]: To name them

File: sharedbuffers/mapped_struct.py (hunk 2 above)
Comment (klaussfreire) [reply sharedbuffers/mapped_struct.py:375]: Nope. FLAG_MASK is a mask to get all flags, WRAP_MASK is the "wrapped" flag only

File: sharedbuffers/mapped_struct.py (hunk 3 above)
Comment (klaussfreire) [reply sharedbuffers/mapped_struct.py:557]: Yes, considerably. Also, not exactly the same. This doesn't accept subclasses, and in many cases we don't want to accept subclasses.

Comment (klaussfreire) [reply sharedbuffers/mapped_struct.py:557]: It would be counterproductive. Check the generated code.

**Pull request closed without merging**